packages/hdwallet-coinbase/src/adapter.ts

```diff
@@ -0,0 +1,61 @@
+import CoinbaseWalletSDK from "@coinbase/wallet-sdk";
+import * as core from "@shapeshiftoss/hdwallet-core";
+
+import { CoinbaseHDWallet } from "./coinbase";
+
+export class CoinbaseAdapter {
+  keyring: core.Keyring;
+
+  private constructor(keyring: core.Keyring) {
+    this.keyring = keyring;
+  }
+
+  public static useKeyring(keyring: core.Keyring) {
+    return new CoinbaseAdapter(keyring);
+  }
+
+  public async initialize(): Promise<number> {
+    return Object.keys(this.keyring.wallets).length;
+  }
+
+  public async pairDevice(): Promise<CoinbaseHDWallet | undefined> {
+    console.info("coinbase-wallet: pairDevice");
+    const APP_NAME = "ShapeShift";
+    const APP_LOGO_URL = "https://avatars.githubusercontent.com/u/52928763?s=50&v=4";
+    const DEFAULT_ETH_JSONRPC_URL = `https://mainnet.infura.io/v3/6e2f28ff4f5340fdb0db5da3baec0af2`;
```

Comment:
can we point this at `https://daemon.ethereum.shapeshift.com`?

Reply:
haha crossed paths. added CoinbaseProviderConfig and pass from web (borrowed from WalletConnect). Should be able to point it anywhere you like via `.env`.